src/NHibernate/Logging.cs

```diff
@@ -63,15 +63,15 @@ private static ILoggerFactory GetLoggerFactory(string nhibernateLoggerClass)
 			}
 			catch (MissingMethodException ex)
 			{
-				throw new ApplicationException("Public constructor was not found for " + loggerFactoryType, ex);
+				throw new Exception("Public constructor was not found for " + loggerFactoryType, ex);
```

Comment:
I would rather throw an ArgumentException here (or InvalidOperationException). We should not throw the base Exception class.